Exercise/Exercise.UnitTests/FileTests.cs

```
@@ -0,0 +1,57 @@
+﻿using System.IO;
+using System.IO.Abstractions;
+using Moq;
+
+namespace Exercise.UnitTests;
+
+[TestClass]
+public class FileTests
+{
+    [TestMethod]
+    public void Ctor_FileDoesNotExist_HasDebugLog()
```

I wasn't quite sure how to test or infact, handle the fact that for example the file does not exist. Should I throw and exception in File.cs when it tries loading it? Or should I just replace the Debug.Write with Logger.Log once possible?

---

Testing: the "safe" thing to do from a test point of view is to use a `Mock<IFileSystem>` and set it up to throw an exception. That way the test is completely in control of the environment, and you know the test is passing for the correct reason.

"Should I throw an exception in File.cs when it tries loading it?": it depends on what you want the user experience to be in the end. Suppressing the error here means the rest of the code will continue to run against an empty file.
That might or might not cause errors later, and might cause a strange user experience e.g. the analysis will run, won't find any issues, and the user will see a message saying "no issues found", which is a bit misleading.
It's almost always better to stop processing when it doesn't make sense to continue.

Here, I wouldn't catch the exception - I'd let it escape for now.
We can always change the error handling later if necessary.

You can still write a test to show that IO errors are not caught.